spec/index.bs

```diff
@@ -6320,11 +6371,20 @@ GPUQueue includes GPUObjectBase;
 
             **Returns:** {{undefined}}
 
-            If any of the following conditions are unsatisfied, generate a validation error and stop.
-            <div class=validusage>
-                - Every {{GPUBuffer}} referenced in any element of |commandBuffers| is in the
-                    `"unmapped"` [=buffer state=].
-                - Every [=usage scope=] contained in |commandBuffers| satisfies the [=usage scope validation=].
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, generate a validation error and stop.
+                    <div class=validusage>
+                        - Every {{GPUBuffer}} referenced in any element of |commandBuffers| is in the
+                            `"unmapped"` [=buffer state=].
+                        - Every [=usage scope=] contained in |commandBuffers| satisfies the [=usage scope validation=].
+                    </div>
+
+                1. Issue the following steps on the [=Queue timeline=] of |this|:
```

Comment:
similarly here, can we make the timeline blocks near each other instead of one being inside of the other?

Reply:
Can we clarify this just a bit? My first instinct would be to format like so:

```
1. Issue the following steps on the [=Device timeline=] of |this|:
<div class=device-timeline>
    1. If any of the following conditions are unsatisfied, generate a validation error and stop.
        <div class=validusage>...</div>
</div>
1. Issue the following steps on the [=Queue timeline=] of |this|:
<div class=queue-timeline>
    1. For each |commandBuffer| in |commandBuffers|:
        1. Execute each command in |commandBuffer|.{{GPUCommandBuffer/[[command_list]]}}.
</div>
```

But to me that reads as though the two timeline's actions are supposed to execute in parallel, when what we want is to complete the validation on the device timeline and then issue the queue timeline steps. I get why you'd like to avoid excessive nesting but the nested did at least imply that timing. An alternative is:

```
1. Issue the following steps on the [=Device timeline=] of |this|:
<div class=device-timeline>
    1. If any of the following conditions are unsatisfied, generate a validation error and stop.
        <div class=validusage>...</div>
</div>
1. If no validation errors were generated after the above steps complete, issue the following steps on the [=Queue timeline=] of |this|:
<div class=queue-timeline>
    1. For each |commandBuffer| in |commandBuffers|:
        1. Execute each command in |commandBuffer|.{{GPUCommandBuffer/[[command_list]]}}.
</div>
```

Which is a clunky but strongly defines the timing of the steps.

Reply:
On a second thought, I agree that the current form of the algorithm is good.
I was thinking about the previous case, where the content timeline and the device timeline were seemingly independent. But still it makes sense to describe it as one launching the other, so I withdraw this concern :)